cegui/src/IconvStringTranscoder.cpp

```diff
@@ -47,7 +47,7 @@ class IconvHelper
         d_toCode(tocode),
         d_cd(iconv_open(d_toCode.c_str(), d_fromCode.c_str()))
     {
-        if (d_cd == reinterpret_cast<iconv_t>(-1))
+        if (d_cd == (iconv_t)(-1))
```

Comment:
`No C-style cast` convention is like a general rule to eliminate all worst cases possible resulting from that cast. This seems to be a static and known case of casting -1. I'm not sure `static_cast` would work on his system particularly. I don't have this compile error anyways. C-style cast in this case `may` be the most portable since it's C library and its developers use it without problems I presume. But yeah I agree that if `static_cast` works, it should be used instead.

Reply:
@Ident It seems that static cast fixes that as well. The reason why I used C-style cast is that I am mostly working with pure C therefore I've not thought about c++ cast. I'm gonna close those MRs and open the new ones.

Reply:
@Ident Instead of creating the new MRs I have just updated the existing ones.

Reply:
@manhnt9 @Ident That's good that MR is still open because it turned out that static cast doesn't fix this issue unfortunately. I have not seen that I used cached version therefore I thought that we can switch from C cast into static cast. I've force pushed this MR once again to use C cast. Here is a toolchain which I use to reproduce this error: http://autobuild.buildroot.org/toolchains/tarballs/br-arm-cortex-a9-glibc-2020.02.tar.bz2